pvlib/pvarray.py

```diff
@@ -0,0 +1,220 @@
+"""
+This module contains implementations of PV module and array electrical models.
+
+These models are used to predict the electrical behavior of pv modules
+or collections of pv modules (arrays).  The primary inputs are
+effective irradiance and operating temperature and the outputs may range from
+power or efficiency at the maximum power point to complete IV curves.
+Supporting functions and parameter fitting functions may also be found here.
+"""
+
+import numpy as np
+from scipy.optimize import curve_fit
+from scipy.special import exp10
+
+
+def pvefficiency_adr(effective_irradiance, temp_cell,
+                     k_a, k_d, tc_d, k_rs, k_rsh):
+    '''
+    Calculate PV module efficiency using the ADR model.
+
+    The efficiency varies with irradiance and operating temperature
+    and is determined by 5 model parameters as described in [1]_.
+
+    Parameters
+    ----------
+    effective_irradiance : numeric, non-negative
+        The effective irradiance incident on the PV module. [W/m²]
+
+    temp_cell : numeric
+        The PV module operating temperature. [°C]
+
+    k_a : numeric
+        Absolute scaling factor, which is equal to the efficiency at
+        reference conditions. This factor allows the model to be used
+        with relative or absolute efficiencies, and to accommodate data sets
+        which are not perfectly normalized but have a slight bias at
+        the reference conditions. [unitless|%]
```

Comment:
```suggestion
        the reference conditions. [unitless|%]
```

I don't see the advantage of allowing either a simple fraction or percent, rather than selecting one or the other. If we keep both, then the Return eta needs to state that it is in the same unit as `k_a`

Reply:
I'll take the option requiring least explanation: unitless.